articles/introduction-to-reinforcement-learning/index.md

```diff
@@ -0,0 +1,205 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/introduction-to-reinforcement-learning/
+title: Understanding Reinforcement Learning
+description:
+author: willies-ogola
+date: 2020-11-06T00:00:00-12:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/introduction-to-reinforcement-learning/hero.png
+    alt: reinforcement learning
+---
+This article will serve as an introduction to Reinforcement Learning (RL). According to [Wikipedia](https://en.wikipedia.org/wiki/Reinforcement_learning), RL is a sub-field of Machine Learning (ML). That is concerned on how [agents](https://en.wikipedia.org/wiki/Software_agent) take actions in an environment to maximize a cumulative reward.
+<!--more-->
+
+### Table of Contents
+1. [Prerequisites](#Prerequisites)
+2. [Key Concepts](#Key-Concepts)
+3. [Introduction](#Introduction)
+4. [Most used Reinforcement Learning Algorithms](#Most-used-Reinforcement-Learning-Algorithms)
+5. [Applications of Reinforcement Learning](#Applications-of-Reinforcement-Learning)
+6. [Final Thoughts](#Final-Thoughts)
+7. [References](#References)
+
+### Prerequisites
+There are two main prerequisites a reader would need before diving into this article:
+- A basic understanding on Machine learning.
+- Some background Knowledge in Linear Algebra, Multivariable Calculus, and Probability and Statistics.
+
+### Key Concepts
+In RL we have key concepts or terms that we need to define. These include agents, state, action, rewards, and the environment.
+
+1.	**Agent**
+The agent decides on what action to take, given an observation from the environment.
+2.	**State**
+The state represents the decision-making factors under consideration observed by the agent.
+3.	**Action**
+The agent selects an action that may change or affect the state and reward.
+4.	**Reward**
+A reward represents the gains and losses in network performance. The reward is for taking a particular action on a particular state in the previous time instance.
```

Comment:
This sentence was not clear:
"The reward is for taking a particular action on a particular state in the previous time instance"

The reward is given? 
I think its saying that a reward is given to an agent when they perform a particular action on a particular state in a better (or shorter amount) of time???
Not sure